### ISSUE TITLE
Loading a checkpoint that was saved in PL < 1.2 still breaks

### ISSUE THREAD

**Issue by ryanking13** [PyTorchLightning/pytorch-lightning#7400]
## 🐛 Bug

Related Issue/PRs.
- #6549
- #6898
- #7031
- #7043

A checkpoint that was saved in PL < 1.2 with `Trainer.add_argparse_args` **still** breaks, when loaded in PL >= 1.2

```sh
AttributeError: Can't get attribute '_gpus_arg_default' on <module 'pytorch_lightning.utilities.argparse_utils' from 
```

### To Reproduce

Use following [**BoringModel**](https://colab.research.google.com/drive/1HvWVVTK8j2Nj52qU4Q4YCyzOm0_aLQF3?usp=sharing) and post here

```python
import argparse

import torch
from torch.utils.data import DataLoader, Dataset

from pytorch_lightning import Trainer, LightningModule


class RandomDataset(Dataset):

    def __init__(self, size, length):
        self.len = length
        self.data = torch.randn(length, size)

    def __getitem__(self, index):
        return self.data[index]

    def __len__(self):
        return self.len


class BoringModel(LightningModule):

    def __init__(self, **kwargs):
        super().__init__()
        self.save_hyperparameters()
        self.layer = torch.nn.Linear(32, 2)

    def forward(self, x):
        return self.layer(x)

    def loss(self, batch, prediction):
        # An arbitrary loss to have a loss that updates the model weights during `Trainer.fit` calls
        return torch.nn.functional.mse_loss(prediction, torch.ones_like(prediction))

    def step(self, x):
        x = self(x)
        out = torch.nn.functional.mse_loss(x, torch.ones_like(x))
        return out

    def training_step(self, batch, batch_idx):
        output = self(batch)
        loss = self.loss(batch, output)
        return {"loss": loss}

    def validation_step(self, batch, batch_idx):
        output = self(batch)
        loss = self.loss(batch, output)
        return {"x": loss}

    def test_step(self, batch, batch_idx):
        output = self(batch)
        loss = self.loss(batch, output)
        return {"y": loss}

    def configure_optimizers(self):
        optimizer = torch.optim.SGD(self.layer.parameters(), lr=0.1)
        lr_scheduler = torch.optim.lr_scheduler.StepLR(optimizer, step_size=1)
        return [optimizer], [lr_scheduler]

    def train_dataloader(self):
        return DataLoader(RandomDataset(32, 64))

    def val_dataloader(self):
        return DataLoader(RandomDataset(32, 64))

    def test_dataloader(self):
        return DataLoader(RandomDataset(32, 64))


if __name__ == '__main__':
    parser = argparse.ArgumentParser()
    parser = Trainer.add_argparse_args(parser)
    args = parser.parse_args()
    model = BoringModel(**vars(args))
    trainer = Trainer.from_argparse_args(args, fast_dev_run=True)
    trainer.fit(model)
    
    # comment the line on >= 1.2.0
    trainer.save_checkpoint('model.pt')
    
    # comment the line below on < 1.2.0
    BoringModel.load_from_checkpoint('model.pt')
```

### Expected behavior

The model should load without any error.

### Environment

**Note**: `Bugs with code` are solved faster ! `Colab Notebook` should be made `public` !

* `IDE`: Please, use our python [bug_report_model.py](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pl_examples/bug_report_model.py
) template.

* `Colab Notebook`: Please copy and paste the output from our [environment collection script](https://raw.githubusercontent.com/PyTorchLightning/pytorch-lightning/master/tests/collect_env_details.py) (or fill out the checklist below manually).

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/PyTorchLightning/pytorch-lightning/master/tests/collect_env_details.py
# For security purposes, please check the contents of collect_env_details.py before running it.
python collect_env_details.py
```

 - PyTorch Version (e.g., 1.0): 1.8.1
 - OS (e.g., Linux): Windows10
 - How you installed PyTorch (`conda`, `pip`, source): pip
 - Build command you used (if compiling from source):
 - Python version: 3.8.6
 - CUDA/cuDNN version: 11.0 / 8.0.2
 - GPU models and configuration: RTX 3070
 - Any other relevant information:

**Comment by awaelchli** [comment]
Which PL version did you load with here?

**Comment by ryanking13** [comment]
I saved a checkpoint with `pytorch-lighting==1.1.8`, and tried to load with `pytorch-lighting==1.2.10`

**Comment by awaelchli** [comment]
yes, the fix didn't make it into 1.2.10. You need to use 1.3.0rc2 last time I checked. Can you try? 
If it still breaks I'm out of words

**Comment by ryanking13** [comment]
Yes, it is a different issue which only happens where checkpoints were saved BEFORE 1.2.0. Sorry for the confusion.
